api/endpoint.go

```diff
@@ -290,6 +290,26 @@ func (e *Endpoint) attributes(ctx *gin.Context, request AttributeRequest) {
 	writeResponse(ctx, metadata, data)
 }
 
+/* The parseSas function will parse the SAS token from the URL and
+ * assign it to the request struct attribute
+ */
+
+func parseSas(vds *string, sas *string) error {
+
+	blob := strings.Split(strings.TrimSpace(*vds), "?")
+	*vds = blob[0]
+
+	if strings.TrimSpace(*sas) != "" {
+		return nil
+	}
+
+	if len(blob) < 2 {
+		return fmt.Errorf("Error:Field validation for 'Sas' parameter %s", *vds)
+	}
+	*sas = blob[1]
```

Comment:
Yes, we assume the user will provide SAS as the following parameter. In case of a separate SAS toke supplied by the user, the function will use the separately provided token.

Also, if the user does not provide the toke in the correct order, it will fail in the later stage and throw an error.

Reply:
I mean that we can for example have a string

`https://account.blob.core.windows.net/container?best=cat&sp=r&st=2023-05-22T12:46:54Z&se=2023-05-22T20:46:54Z&spr=https&sv=2022-11-02&sr=c&sig=signature&timeout=1&versionid=2003-05-22`

openvds would error on this request due to bad value of `versionid` parameter. Without `versionid` parameter call returns correctly.

When user must provide Sas separately and vds separately, it is on user to set 
`SAS="sp=r&st=2023-05-22T12:46:54Z&se=2023-05-22T20:46:54Z&spr=https&sv=2022-11-02&sr=c&sig=signature"` and ignore "best" and "versionid" parameters. If they do not - their problem. We asked you for SAS!

But when user provides us full URL, should we try to parse the whole user input and find out which of provided parameters make SAS, thus throwing away "best" and "versionid" parameters?

Should we be allowed to throw away any user parameters? If we for example throw away "snapshot" - which is [one of the valid](https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob?tabs=azure-ad#uri-parameters) Azure Blob Storage parameters - we will return user incorrect data, not the one they probably intended to get.

On the other hand, "snapshot" or "versionid" or "best" is not a part of sas. So why would we be storing it under the name of sas?? Shouldn't we then at least rename sas parameter to "queryString"?

So wanted to hear your opinion on what is the best approach here :)

Reply:
The Azure blob container generates the vds URL only with SAS toke in exactly the format we are accepting but with no other fields. So, there is no reason to do parsing for any other parameter.  
Even in the worst-case scenario if someone intentionally adds some other parameter it will return the request response.

Reply:
You are very optimistic :smile: 

Worst-case scenario is not exactly true - snapshot is the danger, for example, as we ignore it in cache. Meaning that users might get the wrong data if the first person who created the cache sent request with a snapshot parameter, it got stored in cache and everyone else gets the snapshot version regardless of them providing good sas.

So having severe problems here is possible, even if very unlikely as we are our customers and we probably don't use snapshots.
(Still not 100% sure it is impossible to get data user should have no access to, but can't think of the case.)


Anyway we can't just put our heads in the sand and ignore the problem altogether as it has potential for consequences.
Due to fix complexity and unlikeliness of something going wrong, the absolute minimum we must do is to make user responsible for it by providing some info in docs. Aka refer to the comment somewhere below :slightly_smiling_face:

Reply:
We agree that I will mention in the documentation that the user should provide the URL in the format we accept. Otherwise, it will fail the validation and that is the user's responsibility.